src/models/GitIndex.js

```diff
@@ -201,7 +201,7 @@ export class GitIndex {
     let body = Buffer.concat(
       this.entries.map(entry => {
         // the fixed length + the filename + at least one null char => align by 8
-        let length = Math.ceil((62 + entry.path.length + 1) / 8) * 8
+        let length = Math.ceil((62 + Buffer.from(entry.path).length + 1) / 8) * 8
```

Comment:
in the same way https://github.com/isomorphic-git/isomorphic-git/pull/356/files#diff-1d8985451431c3a96937df9aa85463e7R220